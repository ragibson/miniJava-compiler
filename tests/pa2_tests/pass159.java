// PA1 parse refs pass
class Test {

    int [] a;
    Test [] t;

    int p() {
        t[e] = this + 1;
    }
}

