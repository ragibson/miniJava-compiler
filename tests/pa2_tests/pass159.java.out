********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "Test" classname
  .   FieldDeclList [2]
  .   . (public) FieldDecl
  .   .   ArrayType
  .   .     INT BaseType
  .   .   "a" fieldname
  .   . (public) FieldDecl
  .   .   ArrayType
  .   .     ClassType
  .   .       "Test" Identifier
  .   .   "t" fieldname
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   INT BaseType
  .   .   "p" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [1]
  .   .   . AssignStmt
  .   .   .   IxRef
  .   .   .     IdRef
  .   .   .       "t" Identifier
  .   .   .     RefExpr
  .   .   .       IdRef
  .   .   .         "e" Identifier
  .   .   .   BinaryExpr
  .   .   .     "+" Operator
  .   .   .       RefExpr
  .   .   .         ThisRef
  .   .   .       LiteralExpr
  .   .   .         "1" IntLiteral
=============================================
